modules/nlu/src/views/lite/intentEditor/ContextSelector.tsx

```diff
@@ -0,0 +1,72 @@
+import { Colors, Icon, MenuItem, Position, Tooltip } from '@blueprintjs/core'
+import { ItemRenderer, MultiSelect } from '@blueprintjs/select'
+import React, { FC, useEffect, useState } from 'react'
+
+import { NLUAPI } from '../../api'
+
+interface Props {
+  contexts: string[]
+  api: NLUAPI
+  saveContexts: (ctx: string[]) => void
+}
+
+export const ContextSelector: FC<Props> = props => {
+  const [availableContexts, setContexts] = useState([])
+
+  useEffect(() => {
+    props.api.fetchContexts().then(setContexts)
+  }, [])
+
+  const removeCtx = (ctx: string, idx: number) => {
+    props.saveContexts([...props.contexts.slice(0, idx), ...props.contexts.slice(idx + 1)])
+  }
+
+  const addCtx = (ctx: string) => {
+    if (availableContexts.indexOf(ctx) === -1) {
+      setContexts([...availableContexts, ctx])
+    }
+    props.saveContexts([...props.contexts, ctx])
+  }
+
+  const onItemSelect = (ctx: string) => {
+    const idx = props.contexts.indexOf(ctx)
+    if (idx !== -1) {
+      removeCtx(ctx, idx)
+    } else {
+      addCtx(ctx)
+    }
+  }
+
+  const ctxItemRenderer: ItemRenderer<string> = (ctx, { handleClick, modifiers }) => (
+    <MenuItem
+      text={ctx}
+      key={ctx}
+      onClick={handleClick}
+      active={modifiers.active}
+      icon={props.contexts.indexOf(ctx) !== -1 ? 'tick' : 'blank'}
+    />
+  )
+
+  return (
+    <div>
+      <div>
+        <label htmlFor="selectContext">Contexts</label>
+        &nbsp;
+        <Tooltip content="You can type in the select bar to add new contexts." position={Position.RIGHT}>
+          <Icon color={Colors.GRAY2} icon="info-sign" />
+        </Tooltip>
+      </div>
+      <MultiSelect
+        fill={true}
+        placeholder="Select context..."
+        items={availableContexts}
+        itemRenderer={ctxItemRenderer}
+        onItemSelect={onItemSelect}
+        tagRenderer={ctx => ctx}
+        tagInputProps={{ tagProps: { minimal: true }, onRemove: removeCtx }}
+        popoverProps={{ minimal: true, fill: true }}
+        selectedItems={props.contexts}
```

Comment:
It's missing createNewItemFromQuery / createNewItemRenderer to allow creating new context, right now you can only select existing one (so the tooltip is misleading). 

Maybe we could set a bigger base width for the context field? I think it would look better

![image](https://user-images.githubusercontent.com/42552874/63115581-74947800-bf65-11e9-8165-f3a06088372b.png)